core/src/test/java/org/opensearch/sql/expression/datetime/TimeStampDiffTest.java

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.expression.datetime;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.sql.data.model.ExprDateValue;
+import org.opensearch.sql.data.model.ExprDatetimeValue;
+import org.opensearch.sql.data.model.ExprNullValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimeValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.DSL;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ExpressionTestBase;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.function.FunctionProperties;
+
+class TimeStampDiffTest extends ExpressionTestBase {
+
+  //Helper function to create an argument based on a passed in interval type
+  private static ExprValue generateArg(String intervalType,
+                                       String argType,
+                                       LocalDateTime base,
+                                       int added) {
+    LocalDateTime arg;
+    switch (intervalType) {
+      case "MICROSECOND":
+        arg = base.plusNanos(added * 1000);
+        break;
+      case "SECOND":
+        arg = base.plusSeconds(added);
+        break;
+      case "MINUTE":
+        arg = base.plusMinutes(added);
+        break;
+      case "HOUR":
+        arg = base.plusHours(added);
+        break;
+      case "DAY":
+        arg = base.plusDays(added);
+        break;
+      case "WEEK":
+        arg = base.plusWeeks(added);
+        break;
+      case "MONTH":
+        arg = base.plusMonths(added);
+        break;
+      case "QUARTER":
+        arg = base.plusMonths(added * 3);
+        break;
+      case "YEAR":
+        arg = base.plusYears(added);
+        break;
+      default:
+        arg = null;
```

Comment:
do you want `arg` to ever be `null`...?

Reply:
Maybe this can be reworked. It's not too dangerous though, since this is only used in this test file to help generate test data. I'll see what I can do.